provider_config.go

```diff
@@ -0,0 +1,57 @@
+package jwtauth
+
+import (
+	"fmt"
+)
+
+// Provider-specific configuration interfaces
+// All providers must implement the CustomProvider interface, and may implement
+// others as needed.
+
+// ProviderMap is used to map a provider name to its provider type
+var ProviderMap = map[string]CustomProvider{
+	// TODO: remove "empty" provider when actual providers are added
+	"empty": &EmptyProvider{},
+}
+
+// CustomProvider - Any custom provider must implement this interface
+type CustomProvider interface {
+	Initialize(*jwtConfig) error
+	SensitiveKeys() []string
+}
+
+// NewProviderConfig - returns appropriate provider struct if provider_config
+// specified in jwtConfig. The provider map is provider name-to-instance of a
+// CustomProvider.
+func NewProviderConfig(jc *jwtConfig, providerMap map[string]CustomProvider) (CustomProvider, error) {
+	var provider string
+	var ok bool
+	var newCustomProvider CustomProvider
+
+	if len(jc.ProviderConfig) == 0 {
+		return nil, nil
+	}
+	if provider, ok = jc.ProviderConfig["provider"].(string); !ok {
+		return nil, fmt.Errorf("'provider' field not found in provider_config")
+	}
+	newCustomProvider, ok = providerMap[provider]
+	if !ok {
+		return nil, fmt.Errorf("provider %q not found in custom providers", provider)
+	}
+	err := newCustomProvider.Initialize(jc)
+	if err != nil {
+		return nil, fmt.Errorf("error initializing %q provider_config: %s", provider, err)
+	}
```

Comment:
Same comment/request as before; if `err` isn't used beyond this check, I think `if err := [...]; err != nil {` is better here

Reply:
Changed in fe083e8